### ISSUE TITLE
JWT.Decode with JweAlgorithm.ECDH_ES & JweEncryption.A256GCM

### ISSUE THREAD

**Issue by jehrenzweig** [dvsekhvalnov/jose-jwt#29]
I'm having trouble following the C# examples provided to create and decode a token when using elliptical encryption.  I've posted the test file I used up at [`https://gist.github.com/jehrenzweig/95a176f00236be526f2641e19c2582a7`](https://gist.github.com/jehrenzweig/95a176f00236be526f2641e19c2582a7) -- encoding works fine, but when I try to decode the token the following exception is thrown:

> An unhandled exception of type 'System.Security.Cryptography.CryptographicException' occurred in System.Core.dll
> 
> Additional information: Key does not exist.

**Comment by dvsekhvalnov** [comment]
Hi @jehrenzweig ,

decoding ECDH_ES need elliptic curve key with private part (x,y and d). In your sample you are constructing key only from public part (x,y).

Checkout unit test here: https://github.com/dvsekhvalnov/jose-jwt/blob/master/UnitTests/TestSuite.cs#L1324

**Comment by jehrenzweig** [comment]
Ok, great, that makes more sense.  However, I think the documentation in the `README.md` needs to be updated, since it only alludes to the (x,y) info -- and not (x,y,d) -- when describing **ECDH_ES** for both encoding & decoding.

If you agree, I'd be happy to submit a pull request for review.

**Comment by dvsekhvalnov** [comment]
Yeah, i think example in docs about signature verification, not encryption.

I'll update, no worries, anyway working with docs right now.

I'll keep ticket open until docs updated, feel free to post more questions if it still not working properly.

**Comment by jehrenzweig** [comment]
Great, thank you so much!

**Comment by dvsekhvalnov** [comment]
@jehrenzweig , safe to close issue?

**Comment by jehrenzweig** [comment]
Yep, sorry it took so long! =P
